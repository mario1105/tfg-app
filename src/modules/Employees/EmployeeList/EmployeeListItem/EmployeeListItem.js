import React  from 'react'
import * as R from 'ramda'
import {Grid, Button, makeStyles, FormControl, Typography, Input} from '@material-ui/core'

const useStyles = makeStyles(() => ({
    text: {
        color: 'white'
    }
}))

const EmployeeListItem = ({ id, role, parameters, handleOnChange, editMode, handleOnSubmit, handleEditMode, handleOnRemove }) => {
    const classes = useStyles()
    const { firstName, lastName, email, dateOfBirth, mobilePhone, salary } = parameters
    const isAnyValueEmpty = R.any(value => !value.length, [firstName, lastName, email, dateOfBirth, mobilePhone, salary])

    return (
        <Grid data-test-id={`employees-list-item employees-list-item-${id}`} container>
            <Grid item xs align="left" className={classes.text}>
                <Typography data-test-id={`employees-list-item-${id}-first-name`}>{firstName}</Typography>
            </Grid>
            <Grid  item xs align="left" className={classes.text}>
                <Typography data-test-id={`employees-list-item-${id}-last-name`}>{lastName}</Typography>
            </Grid>
            <Grid item xs align="left" className={classes.text}>
                {editMode
                    ? <FormControl  style={{ paddingRight: '38px' }}>
                        <Input
                            name="email"
                            data-test-id={`employees-list-item-${id}-email-input`}
                            inputProps={{ style: { paddingTop: 0 } }}
                            value={email}
                            onChange={handleOnChange}
                        />
                    </FormControl>
                    : <Typography data-test-id={`employees-list-item-${id}-email`}>{email}</Typography>
                }
            </Grid>
            <Grid item xs align="left" className={classes.text}>
                <Typography data-test-id={`employees-list-item-${id}-date-of-birth`}>{dateOfBirth}</Typography>
            </Grid>
            <Grid item xs align="left" className={classes.text}>
                {editMode
                    ? <FormControl  style={{ paddingRight: '38px' }}>
                        <Input
                            name="mobilePhone"
                            data-test-id={`employees-list-item-${id}-mobile-phone-input`}
                            inputProps={{ style: { paddingTop: 0 } }}
                            value={mobilePhone}
                            onChange={handleOnChange}
                        />
                    </FormControl>
                    : <Typography data-test-id={`employees-list-item-${id}-mobile-phone`}>{mobilePhone}</Typography>
                }
            </Grid>
            { role !== 'user' &&
            <>
                <Grid item xs align="left" className={classes.text}>
                    {editMode && role === 'admin'
                        ? <FormControl  style={{ paddingRight: '38px' }}>
                            <Input
                                name="salary"
                                data-test-id={`employees-list-item-${id}-salary-input`}
                                inputProps={{ style: { paddingTop: 0 } }}
                                value={salary}
                                onChange={handleOnChange}
                            />
                        </FormControl>
                        : <Typography data-test-id={`employees-list-item-${id}-salary`}>{salary}â‚¬</Typography>
                    }
                </Grid>

                <Grid item xs align="left" className={classes.text}>
                    <Button
                        data-test-id={`employees-list-item-${id}-${editMode ? 'submit' : 'edit'}`}
                        variant="outlined"
                        style={{ height: '1.5em', width: '4em' }}
                        onClick={editMode ? handleOnSubmit : handleEditMode}
                        disabled={isAnyValueEmpty}
                    >
                        {editMode ? 'Submit' : 'Edit'}
                    </Button>
                    {role === 'admin' &&
                    <Button
                        data-test-id={`employees-list-item-${id}-remove`}
                        variant="outlined"
                        style={{ height: '1.5em', width: '5em', marginLeft: '1em', color: 'orange' }}
                        onClick={handleOnRemove}
                    >
                        Remove
                    </Button>
                    }
                </Grid>
            </>
            }
        </Grid>
    )
}

export default EmployeeListItem
