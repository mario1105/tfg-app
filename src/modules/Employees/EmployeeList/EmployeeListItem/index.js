import React, {useEffect, useState} from 'react'
import EmployeeListItem from "./EmployeeListItem"

const EmployeeListItemContainer = ({ id, role, firstName, lastName, email, dateOfBirth, mobilePhone, salary, handleRemoveEmployee }) => {
    const [parameters, setParameters] = useState({firstName, lastName, email, dateOfBirth, mobilePhone, salary})
    const [editMode, setEditMode] = useState(false)

    const handleOnChange = e => {
        setParameters({
            ...parameters,
            [e.currentTarget.name]: e.currentTarget.value
        })
    }

    const handleEditMode = () => setEditMode(!editMode)

    const handleOnSubmit = async () => {
        const options = {
            headers: {
                Accept: 'application/json',
                'Content-Type': 'application/json'
            },
            method: 'PATCH',
            body: JSON.stringify({
                ...parameters
            })
        }

        await fetch(`http://localhost:3001/employees/${id}`, options)
        handleEditMode()
    }

    const handleOnRemove = async () => {
        const options = {
            headers: {
                Accept: 'application/json',
                'Content-Type': 'application/json'
            },
            method: 'DELETE',
        }

        await fetch(`http://localhost:3001/employees/${id}`, options)
        handleRemoveEmployee(id)
    }

    return (
        <EmployeeListItem
            role={role}
            parameters={parameters}
            editMode={editMode}
            handleOnChange={handleOnChange}
            handleEditMode={handleEditMode}
            handleOnSubmit={handleOnSubmit}
            handleOnRemove={handleOnRemove}
        />
    )
}

export default EmployeeListItemContainer

