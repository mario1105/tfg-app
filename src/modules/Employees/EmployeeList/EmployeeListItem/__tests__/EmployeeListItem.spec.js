import React from 'react'
import { shallow } from 'enzyme'
import EmployeeListItem from "../EmployeeListItem"

const id = 1
const parameters = {
    firstName: 'Mario',
    lastName: 'Jimenez',
    email: 'mario@test.com',
    dateOfBirth: '11/05/1995',
    mobilePhone: '666555444',
    salary: '355000'
}

const getProps = ({ role, editMode }) => ({
    id,
    role,
    editMode,
    parameters,
    handleOnChange: () => null,
    handleOnSubmit: () => null,
    handleEditMode: () => null,
    handleOnRemove: () => null
})

const renderComponent = ({ role, editMode }) => {
    return (shallow(<EmployeeListItem {...getProps({ role, editMode })} />))
}

let wrapper
describe('EmployeeListItem Unit Tests', () => {
    describe('when the user is an admin', () => {
        describe('when the edit mode is disabled', () => {
            beforeEach( () => {
                wrapper = renderComponent({ role: 'admin', editMode: false})
            })
            it('renders as expected', () => {
                expect(wrapper).toMatchSnapshot()
            })
        })
        describe('when the edit mode is enabled', () => {
            beforeEach( () => {
                wrapper = renderComponent({ role: 'admin', editMode: true})
            })
            it('renders as expected', () => {
                expect(wrapper).toMatchSnapshot()
            })
        })
    })
    describe('when the user is an associate', () => {
        describe('when the edit mode is disabled', () => {
            beforeEach( () => {
                wrapper = renderComponent({ role: 'associate', editMode: false})
            })
            it('renders as expected', () => {
                expect(wrapper).toMatchSnapshot()
            })
            it('renders all the fields available for the associate',  () => {
                expect(wrapper.find(`[data-test-id~="employees-list-item-${id}-first-name"]`)
                    .contains(parameters.firstName)).toBe(true)
                expect(wrapper.find(`[data-test-id~="employees-list-item-${id}-last-name"]`)
                    .contains(parameters.lastName)).toBe(true)
                expect(wrapper.find(`[data-test-id~="employees-list-item-${id}-email"]`)
                    .contains(parameters.email)).toBe(true)
                expect(wrapper.find(`[data-test-id~="employees-list-item-${id}-date-of-birth"]`)
                    .contains(parameters.dateOfBirth)).toBe(true)
                expect(wrapper.find(`[data-test-id~="employees-list-item-${id}-mobile-phone"]`)
                    .contains(parameters.mobilePhone)).toBe(true)
                expect(wrapper.find(`[data-test-id~="employees-list-item-${id}-salary"]`)
                    .contains(parameters.salary)).toBe(true)
            })
            it('renders the edit button',  () => {
                expect(wrapper.find(`[data-test-id~="employees-list-item-${id}-edit"]`)).toHaveLength(1)
            })
            it('does not render the remove button',  () => {
                expect(wrapper.find(`[data-test-id~="employees-list-item-${id}-remove"]`)).toHaveLength(0)
            })
        })
        describe('when the edit mode is enabled', () => {
            beforeEach( () => {
                wrapper = renderComponent({ role: 'associate', editMode: true})
            })
            it('renders as expected', () => {
                expect(wrapper).toMatchSnapshot()
            })
            it('renders email and mobile phone inputs',  () => {
                expect(wrapper.find(`[data-test-id~="employees-list-item-${id}-email-input"]`)).toHaveLength(1)
                expect(wrapper.find(`[data-test-id~="employees-list-item-${id}-mobile-phone-input"]`)).toHaveLength(1)
            })
            it('does not render salary input',  () => {
                expect(wrapper.find(`[data-test-id~="employees-list-item-${id}-salary-input"]`)).toHaveLength(0)
            })
            it('renders the submit button',  () => {
                expect(wrapper.find(`[data-test-id~="employees-list-item-${id}-submit"]`)).toHaveLength(1)
            })
        })
    })
    describe('when the user is an user', () => {
        beforeEach( () => {
            wrapper = renderComponent({ role: 'user', editMode: false})
        })
        it('renders as expected', () => {
            expect(wrapper).toMatchSnapshot()
        })
        it('renders all the fields available for the associate',  () => {
            expect(wrapper.find(`[data-test-id~="employees-list-item-${id}-first-name"]`)
                .contains(parameters.firstName)).toBe(true)
            expect(wrapper.find(`[data-test-id~="employees-list-item-${id}-last-name"]`)
                .contains(parameters.lastName)).toBe(true)
            expect(wrapper.find(`[data-test-id~="employees-list-item-${id}-email"]`)
                .contains(parameters.email)).toBe(true)
            expect(wrapper.find(`[data-test-id~="employees-list-item-${id}-date-of-birth"]`)
                .contains(parameters.dateOfBirth)).toBe(true)
            expect(wrapper.find(`[data-test-id~="employees-list-item-${id}-mobile-phone"]`)
                .contains(parameters.mobilePhone)).toBe(true)
        })
        it('does not render the salary field',  () => {
            expect(wrapper.find(`[data-test-id~="employees-list-item-${id}-salary"]`)).toHaveLength(0)
        })
        it('does not render the edit button',  () => {
            expect(wrapper.find(`[data-test-id~="employees-list-item-${id}-edit"]`)).toHaveLength(0)
        })
        it('does not render the remove button',  () => {
            expect(wrapper.find(`[data-test-id~="employees-list-item-${id}-remove"]`)).toHaveLength(0)
        })
    })

})
