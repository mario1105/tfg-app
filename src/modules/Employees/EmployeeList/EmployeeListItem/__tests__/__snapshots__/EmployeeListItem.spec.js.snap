// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeListItem Unit Tests when the user is an admin when the edit mode is disabled renders as expected 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  data-test-id="employees-list-item employees-list-item-1"
>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-first-name"
    >
      Mario
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-last-name"
    >
      Jimenez
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-email"
    >
      mario@test.com
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-date-of-birth"
    >
      11/05/1995
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-mobile-phone"
    >
      666555444
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-salary"
    >
      355000
      €
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Button))
      data-test-id="employees-list-item-1-edit"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "height": "1.5em",
          "width": "4em",
        }
      }
      variant="outlined"
    >
      Edit
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      data-test-id="employees-list-item-1-remove"
      onClick={[Function]}
      style={
        Object {
          "color": "orange",
          "height": "1.5em",
          "marginLeft": "1em",
          "width": "5em",
        }
      }
      variant="outlined"
    >
      Remove
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`EmployeeListItem Unit Tests when the user is an admin when the edit mode is enabled renders as expected 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  data-test-id="employees-list-item employees-list-item-1"
>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-first-name"
    >
      Mario
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-last-name"
    >
      Jimenez
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(FormControl))
      style={
        Object {
          "paddingRight": "38px",
        }
      }
    >
      <WithStyles(ForwardRef(Input))
        data-test-id="employees-list-item-1-email-input"
        inputProps={
          Object {
            "style": Object {
              "paddingTop": 0,
            },
          }
        }
        name="email"
        onChange={[Function]}
        value="mario@test.com"
      />
    </WithStyles(ForwardRef(FormControl))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-date-of-birth"
    >
      11/05/1995
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(FormControl))
      style={
        Object {
          "paddingRight": "38px",
        }
      }
    >
      <WithStyles(ForwardRef(Input))
        data-test-id="employees-list-item-1-mobile-phone-input"
        inputProps={
          Object {
            "style": Object {
              "paddingTop": 0,
            },
          }
        }
        name="mobilePhone"
        onChange={[Function]}
        value="666555444"
      />
    </WithStyles(ForwardRef(FormControl))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(FormControl))
      style={
        Object {
          "paddingRight": "38px",
        }
      }
    >
      <WithStyles(ForwardRef(Input))
        data-test-id="employees-list-item-1-salary-input"
        inputProps={
          Object {
            "style": Object {
              "paddingTop": 0,
            },
          }
        }
        name="salary"
        onChange={[Function]}
        value="355000"
      />
    </WithStyles(ForwardRef(FormControl))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Button))
      data-test-id="employees-list-item-1-submit"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "height": "1.5em",
          "width": "4em",
        }
      }
      variant="outlined"
    >
      Submit
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      data-test-id="employees-list-item-1-remove"
      onClick={[Function]}
      style={
        Object {
          "color": "orange",
          "height": "1.5em",
          "marginLeft": "1em",
          "width": "5em",
        }
      }
      variant="outlined"
    >
      Remove
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`EmployeeListItem Unit Tests when the user is an associate when the edit mode is disabled renders as expected 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  data-test-id="employees-list-item employees-list-item-1"
>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-first-name"
    >
      Mario
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-last-name"
    >
      Jimenez
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-email"
    >
      mario@test.com
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-date-of-birth"
    >
      11/05/1995
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-mobile-phone"
    >
      666555444
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-salary"
    >
      355000
      €
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Button))
      data-test-id="employees-list-item-1-edit"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "height": "1.5em",
          "width": "4em",
        }
      }
      variant="outlined"
    >
      Edit
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`EmployeeListItem Unit Tests when the user is an associate when the edit mode is enabled renders as expected 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  data-test-id="employees-list-item employees-list-item-1"
>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-first-name"
    >
      Mario
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-last-name"
    >
      Jimenez
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(FormControl))
      style={
        Object {
          "paddingRight": "38px",
        }
      }
    >
      <WithStyles(ForwardRef(Input))
        data-test-id="employees-list-item-1-email-input"
        inputProps={
          Object {
            "style": Object {
              "paddingTop": 0,
            },
          }
        }
        name="email"
        onChange={[Function]}
        value="mario@test.com"
      />
    </WithStyles(ForwardRef(FormControl))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-date-of-birth"
    >
      11/05/1995
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(FormControl))
      style={
        Object {
          "paddingRight": "38px",
        }
      }
    >
      <WithStyles(ForwardRef(Input))
        data-test-id="employees-list-item-1-mobile-phone-input"
        inputProps={
          Object {
            "style": Object {
              "paddingTop": 0,
            },
          }
        }
        name="mobilePhone"
        onChange={[Function]}
        value="666555444"
      />
    </WithStyles(ForwardRef(FormControl))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-salary"
    >
      355000
      €
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Button))
      data-test-id="employees-list-item-1-submit"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "height": "1.5em",
          "width": "4em",
        }
      }
      variant="outlined"
    >
      Submit
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`EmployeeListItem Unit Tests when the user is an user renders as expected 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  data-test-id="employees-list-item employees-list-item-1"
>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-first-name"
    >
      Mario
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-last-name"
    >
      Jimenez
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-email"
    >
      mario@test.com
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-date-of-birth"
    >
      11/05/1995
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    align="left"
    className="makeStyles-text-1"
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      data-test-id="employees-list-item-1-mobile-phone"
    >
      666555444
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
